src/views/CoCurricularTranscript/Components/Activity/index.tsx

```diff
@@ -66,3 +80,24 @@ const formatDuration = (sessionRecords: MembershipHistorySession[]) => {
 
   return intervalDescriptions.join(', ');
 };
+
+const formatFinalDuration = (sessionRecords: MembershipHistorySession[]) => {
+  const sessions = sessionRecords
+    .map((s) => sessionService.parseSessionCode(s.SessionCode))
+    .sort(compareAsc);
+
+  const intervalDescriptions: string[] = [];
+  let interval: MembershipInterval | undefined;
+
+  sessions.forEach((session) => {
+    if (interval === undefined) {
+      // If this is the first session, initialize interval.
+      interval = new MembershipInterval(session);
+    } else {
+      // Extend the interval.
+      interval.extendTo(session);
+    }
+  });
+
+  return interval?.toString();
+};
```

Comment:
This is a lot of repeated code for doing something only slightly different from `formatDuration(sessions)`.

Perhaps you could eliminate `formatFinalDuration(...)` and update the `formatDuration(sessions)` function to the following:

```
const formatDuration = (sessionRecords: MembershipHistorySession[], summarizeDates: bool = false) => {
...
    } else if (interval.consecutiveWith(session) || summarizeDates) {
...
  return intervalDescriptions.join(', ');
};
```

Then you could just call `formatDuration(sessions, true)` instead of `formatFinalDuration(...)`